### PR TITLE
Add github action to build binaries for each platform at release

### PR DESCRIPTION
**Pull request by s1061123**
This change introduces new github action to build cni plugin binaries for each platform at tagged new release.

### CODE REVIEW

**Comment by s1061123** [comment]
Here is the sample release with my temporaly cloned repo with PR code: https://github.com/s1061123/plugins-test-repo/releases

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+jobs:
+  linux_release:
+    name: Release linux binaries
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goarch: [amd64, arm, arm64, mips64le, ppc64le, riscv64, s390x]
+    steps:
+      - name: Install Go
```
[the remark this answers is not in the document]
Comment (s1061123) [reply .github/workflows/release.yaml:16]: I understand but it means we use go 1.15 as following:
https://github.com/containernetworking/plugins/blob/b6a0e0bc96906f0d3bd6bfcaab0b5ae72292f46c/.github/workflows/test.yaml#L7

Based on [golang releases page](https://go.dev/doc/devel/release), latest golang1.15, 1.15.15, was released 2years+ ago.
Golang community supports the last two major versions, based on 'Release Policy' section of release page, hence now golang supports go1.20 and go1.21.

Should we update golang to tests as well as builds? (If you don't mind it, I will update test golang in this PR)

**Comment by s1061123** [comment]
@squeed Before merge the PR, you may need to change repository config change as following:

"Setting (in Repositories)" -> "Actions (General)" -> "Workflow permissions": Change it to 'Read and write permissions"

File: .github/workflows/release.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (squeed) [reply .github/workflows/release.yaml:16]: yes, definitely! Thanks!

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        run: cp README.md LICENSE bin/
+
+      - name: Create dist directory
+        run: mkdir dist
+
+      - name: Create archive file
+        working-directory: ./bin
+        run: tar cfzv ../dist/cni-plugins-linux-${{ matrix.goarch }}-${{ github.ref_name }}.tgz .
+
+      - name: Create sha1 checksum
```
Comment (squeed) [inline .github/workflows/release.yaml:40]: You know, we should probably stop producing sha1 checksums :-)

**Comment by squeed** [comment]
@s1061123 this is awesome! Thanks so much! Let's figure out the right way to set the go version and get this merged.

File: .github/workflows/release.yaml (hunk 2 above)
Comment (s1061123) [reply .github/workflows/release.yaml]: Removed.

**Comment by s1061123** [comment]
@squeed Regarding shared variables in github action, we may use 'configuration variables' in github, available at repository config web UI.

https://github.blog/changelog/2023-01-10-github-actions-support-for-configuration-variables-in-workflows/

This requires to change repository setting, so I cannot care that. Please change web UI and this PR (you can pull my branch and you can modify the branch directly as 'Allow edits and access to secrets by maintainers' enabled)

**Pull request merged**